mypy/binder.py

```diff
@@ -41,12 +41,22 @@ class Frame:
     """A Frame represents a specific point in the execution of a program.
     It carries information about the current types of expressions at
     that point, arising either from assignments to those expressions
-    or the result of isinstance checks. It also records whether it is
-    possible to reach that point at all.
+    or the result of isinstance checks and other type narrowing
+    operations. It also records whether it is possible to reach that
```

Comment:
You can also mention that `isinstance()` etc. follow a more direct path via `put()`, while assignments have some additional special-casing and first go via `assign_type()`.